src/main/java/redis/clients/jedis/Protocol.java

```diff
@@ -274,17 +273,18 @@ public static Object read(final RedisInputStream is) {
   }
 
   public static enum Keyword {
-    AGGREGATE, ALPHA, ASC, BY, DESC, GET, LIMIT, MESSAGE, NO, NOSORT, PMESSAGE, PSUBSCRIBE,
-    PUNSUBSCRIBE, OK, ONE, QUEUED, SET, STORE, SUBSCRIBE, UNSUBSCRIBE, WEIGHTS, WITHSCORES,
+    // Request keywords
+    AGGREGATE, ALPHA, ASC, BY, DESC, GET, LIMIT, NO, NOSORT, ONE, SET, STORE, WEIGHTS, WITHSCORES,
     RESETSTAT, REWRITE, RESET, FLUSH, EXISTS, LOAD, KILL, LEN, REFCOUNT, ENCODING, IDLETIME,
-    GETNAME, SETNAME, LIST, MATCH, COUNT, PING, PONG, UNLOAD, REPLACE, KEYS, PAUSE, DOCTOR, 
-    BLOCK, NOACK, STREAMS, KEY, CREATE, MKSTREAM, SETID, DESTROY, DELCONSUMER, MAXLEN, GROUP, 
-    IDLE, TIME, RETRYCOUNT, FORCE;
+    GETNAME, SETNAME, LIST, MATCH, COUNT, UNLOAD, REPLACE, KEYS, PAUSE, DOCTOR, BLOCK, NOACK,
+    STREAMS, CREATE, MKSTREAM, SETID, DESTROY, DELCONSUMER, MAXLEN, GROUP, IDLE, RETRYCOUNT, FORCE,
+    // Response keywords
+    OK, QUEUED, message, pmessage, psubscribe, punsubscribe, subscribe, unsubscribe, pong;
```

Comment:
why do you need to have the keywords in lowercase?

Reply:
Because Redis responds those responses in lower cases.

Reply:
```
$ redis-cli
127.0.0.1:6379> ping
PONG
127.0.0.1:6379>
```

Reply:
I mean, in PUB/SUB.

Reply:
https://github.com/xetorthio/jedis/blob/c4612a379ab8bad627b28193bb44604825f4b4bd/src/main/java/redis/clients/jedis/JedisPubSub.java#L163

Reply:
so won't we have issue with `ping`?

Reply:
Because `ping` is a request and Redis accepts both upper and lower cased letters in case of requests.

Reply:
Sorry I meant `pong`

Reply:
You mean, why we don't have issue with `pong` right now? Because it is lower cased in 
https://github.com/xetorthio/jedis/blob/c370114e84ea0fc27a20c6d2244d150826e31e71/src/main/java/redis/clients/jedis/Protocol.java#L286-L288 

This is dependant on `Locale` which I'm thinking about not using.